lib/core/src/animation/physics.rs

```diff
@@ -0,0 +1,248 @@
+    pub fn set_velocity(&mut self, velocity:Vector3<f32>) {
+        self.data.borrow_mut().velocity = velocity
+    }
+
+    /// Gets acceleration.
+    pub fn acceleration(&self) -> Vector3<f32> {
+        self.data.borrow().acceleration
+    }
+
+    /// Sets acceleration.
+    pub fn set_acceleration(&mut self, acceleration:Vector3<f32>) {
+        self.data.borrow_mut().acceleration = acceleration
+    }
+}
+
+impl HasPosition for KinematicProperties {
+    fn position(&self) -> Vector3<f32> {
+        self.data.borrow().position
+    }
+
+    fn set_position(&mut self, position:Vector3<f32>) {
+        self.data.borrow_mut().position = position
+    }
+}
+
+// ======================
+// === PropertiesData ===
+// ======================
+
+struct PropertiesData {
+    kinematics : KinematicProperties,
+    spring     : SpringProperties,
+    drag       : DragProperties
+}
+
+impl PropertiesData {
+    pub fn new
+    (kinematics:KinematicProperties, spring:SpringProperties, drag:DragProperties) -> Self {
+        Self { kinematics, spring, drag }
+    }
+}
+
+// ==================
+// === Properties ===
+// ==================
+
+/// A structure including kinematics, drag and spring properties.
+#[derive(Clone)]
+pub struct Properties {
+    data : Rc<RefCell<PropertiesData>>
+}
+
+impl Properties {
+    pub fn new
+    (kinematics:KinematicProperties, spring:SpringProperties, drag:DragProperties) -> Self {
+        let data = Rc::new(RefCell::new(PropertiesData::new(kinematics, spring, drag)));
+        Self { data }
+    }
+
+    pub fn kinematics(&self) -> KinematicProperties { self.data.borrow().kinematics.clone() }
+
+    pub fn set_kinematics(&mut self, kinematics:KinematicProperties) {
+        self.data.borrow_mut().kinematics = kinematics
+    }
+    pub fn spring(&self) -> SpringProperties { self.data.borrow().spring.clone() }
+
+    pub fn set_spring(&mut self, spring:SpringProperties) {
+        self.data.borrow_mut().spring = spring
+    }
+    pub fn drag(&self) -> DragProperties { self.data.borrow().drag.clone() }
+
+    pub fn set_drag(&mut self, drag:DragProperties) {
+        self.data.borrow_mut().drag = drag
+    }
+}
+
+// ========================
+// === PhysicsSimulator ===
+// ========================
+
+/// A 60 steps per second physics simulator used to simulate `Properties`.
+pub struct PhysicsSimulator {
+    _animator : Animator
+}
+
+/// Simulate the `KinematicProperties`.
+fn simulate_kinematics(properties:&mut KinematicProperties, dt:f32) {
+    properties.set_velocity(properties.velocity() + properties.acceleration() * dt);
+    properties.set_position(properties.position() + properties.velocity()     * dt);
+}
+
+/// Simulate dragging on `KinematicProperties`.
+fn simulate_dragging(kinematics:&mut KinematicProperties, drag:&DragProperties) {
+    let velocity = kinematics.velocity();
+    let speed    = velocity.norm();
+    kinematics.set_velocity(velocity / (1.0 + drag.amount() * speed));
```

Comment:
I believe that this equation for velocity change is wrong. Where are you using mass property of the object? Moreover, the whole equation is not physicaslly correct.

Reply:
@wdanilo I am lost now. We had a discussion about the new equation I was using which was taking time into consideration and you asked me to use the older one because you said it was physically correct. This is the older one. What should I do?

Reply:
I understand that now the problem is solved and we know how to code it? :)

Reply:
@wdanilo yes, we came to an agreement! :+1: